climpred/prediction.py

```diff
@@ -935,18 +978,49 @@ def bootstrap_perfect_model(ds, control, metric='rmse', comparison='m2m',
 # --------------------------------------------#
 # PREDICTABILITY HORIZON
 # --------------------------------------------#
-def xr_predictability_horizon(skill, threshold, limit='upper'):
+def xr_predictability_horizon(skill, threshold, limit='upper',
+                              p_values=None, N=None, alpha=0.05, ci=90):
     """
     Get predictability horizons of dataset from skill and
     threshold dataset.
+
+    Inputs:
+        skill: (xarray object) skill (e.g., ACC) at different lead times.
+        threshold: (xarray object) skill for persistence or uninitialized
+                   ensemble.
+        limit: (optional str) If 'upper' check horizon for correlation
+               coefficients by testing lead time to which the skill
+               beats out the threshold. If 'lower', check horizon for which
+               error (e.g., MAE) is lower than threshold.
+        p_values: (optional xarray object) If using 'upper' limit, input
+                  a DataArray/Dataset of the same dimensions as skill that
+                  contains p-values for the skill correlatons.
+
+    Returns:
+        predictability horizon reduced by the lead time dimension.
     """
     if limit is 'upper':
-        ph = (skill > threshold).argmin('time')
```

Comment:
One last change request as I would like to keep using xr_ph with my bootstrapping approach also: can we make the arguments p_values and N non-compulsory ? Maybe first check a flag. Then I can still use the function in its original form for the bootstrapping. As long as my notebook for PM still runs I am fine.

Reply:
Yes I will work on this. I will either add a `horizon` keyword to compute_perfect_model that triggers certain keywords in the predictability horizon function, or I'll add something to the horizon function to turn off the required p_values and N. I will test this by either having your notebook run without fail or with minor edits.